### PR TITLE
Improving test

### PR DESCRIPTION
**Pull request by kenichi**
PR for commenting 😺

### CODE REVIEW

File: lib/reel/session/store.rb [hunk 1]
```diff
@@ -20,8 +20,8 @@ def initialize request
             @key = decrypt(array_val[1]) if decrypt(array_val[0]) ==  Reel::Session.configuration[:session_name]
           end
         end
-        # check if key exist in our concurrent hash
-        @val = @store[@key] if @store.key? @key
+        # getting value if key exist in our concurrent hash
+        @val = @store[@key]
```
Comment (kenichi) [inline lib/reel/session/store.rb:24]: 👍

**Comment by kenichi** [comment]
this looks great! again, we should discuss when to not generate/send cookies though. rack sends it all the time, i'm not sure clearing the session hash would be a good case for not setting the cookie.

have you explored what rack does in its different impls? the Cookie one, which marshals a hash straight into the cookie value obviously still sends it, but the Pool one, which is similar to yours, does too. Lemme know your thoughts here.

**Comment by pulkit4tech** [comment]
@kenichi as you rightly said, even if we don't set any header .. still browser will send key in header until expiry(if some session value was set previously). I agree with you and will change it to generated `HeaderError` (as now if key will be `nil` its `erroneous`)

**Comment by pulkit4tech** [comment]
As `tweaking_configuration` logic is almost final with very small tweaks .. I am merging this pull and will update `session#configuration` in `spec` accordingly..

**Pull request merged**